lib/active_model/validations/collection_items_validator.rb

```diff
@@ -0,0 +1,56 @@
+module ActiveModel
+  module Validations
+    # Adds collection items validation to ActiveModel::Model.
+    #
+    # Usage:
+    # @example
+    #   class FooBar
+    #     include ActiveModel::Model
+    #
+    #     attr_accessor :kinds
+    #
+    #     validates :kinds, collection_items { type: Symbol, inclusion: { in: [:foo, :bar] } }
+    #   end
+    class CollectionItemsValidator < ActiveModel::EachValidator
+      extend ActiveSupport::Autoload
+
+      autoload :ArrayAttributeReader
+      autoload :ArrayAttributeGetter
+      autoload :InnerValidatorBuilder
+
+      def initialize(*)
+        super.tap { set_inner_validators }
+      end
+
+      def validate_each(record, attribute, value)
+        if value.kind_of? Enumerable
+          validate_collection(record, attribute, value)
+        else
+          record.errors.add attribute, 'must be a collection'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.